discoverers/discoverer_aws_ec2.go

```diff
@@ -232,16 +268,136 @@ func (ec2d *AwsEc2Discoverer) Discover(ctx context.Context) *discovery.Result {
 
 				InstanceSsmStatus: ssmInstanceStatus,
 			}
-			result.AddResources(discovery.Resource{
-				ResourceType:          discovery.ResourceTypeAwsEc2Instance,
-				AwsEc2InstanceDetails: ec2InstanceDetails,
-			})
+
+			wg.Add(1)
+			go ec2d.reachabilityCheckAndAdd(ctx, &wg, result, ec2InstanceDetails)
 		}
 	}
 
 	return result
 }
 
+func (ec2d *AwsEc2Discoverer) reachabilityCheckAndAdd(
+	ctx context.Context,
+	wg *sync.WaitGroup,
+	result *discovery.Result,
+	ec2Details *discovery.AwsEc2InstanceDetails,
+) {
+	defer wg.Done()
+
+	if ec2d.networkReachabilityCheckEnabled {
+		var testswg sync.WaitGroup
+
+		if ec2Details.PrivateDnsName != "" {
+			testswg.Add(1)
```

Comment:
we can use the https://pkg.go.dev/golang.org/x/sync/errgroup here, to avoid handling all the wg manually